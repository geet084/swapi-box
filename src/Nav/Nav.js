import React from 'react'
import './Nav.scss'
import Icon from '../Icon/Icon'

const baseURL = 'http://icons.iconarchive.com/icons/sensibleworld/starwars/64/';

const Nav = ({ choice, getInfo }) => {
  let iconList = ['people', 'planets', 'vehicles']
  let iconURL = ['Leia-icon.png', 'Death-Star-icon.png', 'Millennium-Falcon-icon.png']

  let icons = iconList.map((icon, i) => {
    let selectedIcon = choice === icon
    let curURL = baseURL + iconURL[i]

    if (selectedIcon) {
      return <Icon key={i} url={curURL} choice={icon} getInfo={getInfo} classes={'active'} />
    } else {
      return <Icon key={i} url={curURL} choice={icon} getInfo={getInfo} classes={'inactive'} />
    }
  })

  return (
    <div className="nav">
      {icons}
  </div>
  )
}

export default Nav;