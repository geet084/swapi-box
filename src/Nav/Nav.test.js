import React from 'react'
import { shallow } from 'enzyme'
import Nav from './Nav'

describe('Nav', () => {
  let wrapper;
  const mockFetchInfo = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Nav fetchInfo={mockFetchInfo}/>
    )
  })

  it('should properly render snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

})