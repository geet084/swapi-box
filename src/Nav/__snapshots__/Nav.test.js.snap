// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should properly render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    fetchInfo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          choice="people"
          classes="inactive"
          url="http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Leia-icon.png"
        />,
        <Icon
          choice="planets"
          classes="inactive"
          url="http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Death-Star-icon.png"
        />,
        <Icon
          choice="vehicles"
          classes="inactive"
          url="http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Millennium-Falcon-icon.png"
        />,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "choice": "people",
          "classes": "inactive",
          "getInfo": undefined,
          "url": "http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Leia-icon.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "choice": "planets",
          "classes": "inactive",
          "getInfo": undefined,
          "url": "http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Death-Star-icon.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "choice": "vehicles",
          "classes": "inactive",
          "getInfo": undefined,
          "url": "http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Millennium-Falcon-icon.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            choice="people"
            classes="inactive"
            url="http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Leia-icon.png"
          />,
          <Icon
            choice="planets"
            classes="inactive"
            url="http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Death-Star-icon.png"
          />,
          <Icon
            choice="vehicles"
            classes="inactive"
            url="http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Millennium-Falcon-icon.png"
          />,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "choice": "people",
            "classes": "inactive",
            "getInfo": undefined,
            "url": "http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Leia-icon.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "choice": "planets",
            "classes": "inactive",
            "getInfo": undefined,
            "url": "http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Death-Star-icon.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "choice": "vehicles",
            "classes": "inactive",
            "getInfo": undefined,
            "url": "http://icons.iconarchive.com/icons/sensibleworld/starwars/64/Millennium-Falcon-icon.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
